felix/nftables/match_builder.go

```diff
@@ -0,0 +1,489 @@
+	return m
+}
+
+func (m nftMatch) NotConntrackState(stateNames string) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("ct state != %s", strings.ToLower(stateNames)))
+	return m
+}
+
+func (m nftMatch) Protocol(name string) generictables.MatchCriteria {
+	if m.proto != "" {
+		logrus.WithField("protocol", m.proto).Fatal("Protocol already set")
+	} else if m.protoNum != 0 {
+		logrus.WithField("protocol", m.protoNum).Fatal("Protocol already set")
+	}
+	m.proto = name
+
+	// The "meta l4proto" matches on nftables metadata about the packet, which allows this
+	// match to work for both IPv4 and IPv6 packets.  The "ip protocol" match only works for
+	// IPv4 packets.
+	m.clauses = append(m.clauses, fmt.Sprintf("meta l4proto %s", name))
+	return m
+}
+
+func (m nftMatch) NotProtocol(name string) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("meta l4proto != %s", name))
+	return m
+}
+
+func (m nftMatch) ProtocolNum(num uint8) generictables.MatchCriteria {
+	if m.proto != "" {
+		logrus.WithField("protocol", m.proto).Fatal("Protocol already set")
+	} else if m.protoNum != 0 {
+		logrus.WithField("protocol", m.protoNum).Fatal("Protocol already set")
+	}
+	m.protoNum = num
+
+	// The "meta l4proto" matches on nftables metadata about the packet, which allows this
+	// match to work for both IPv4 and IPv6 packets.  The "ip protocol" match only works for
+	// IPv4 packets.
+	m.clauses = append(m.clauses, fmt.Sprintf("meta l4proto %s", protoNumToName(num)))
+	return m
+}
+
+func (m nftMatch) NotProtocolNum(num uint8) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("meta l4proto != %s", protoNumToName(num)))
+	return m
+}
+
+func (m nftMatch) SourceNet(net string) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("IPV saddr %s", net))
+	return m
+}
+
+func (m nftMatch) NotSourceNet(net string) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("IPV saddr != %s", net))
+	return m
+}
+
+func (m nftMatch) DestNet(net string) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("IPV daddr %s", net))
+	return m
+}
+
+func (m nftMatch) NotDestNet(net string) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("IPV daddr != %s", net))
+	return m
+}
+
+func (m nftMatch) SourceIPSet(name string) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("IPV saddr @%s", LegalizeSetName(name)))
+	return m
+}
+
+func (m nftMatch) NotSourceIPSet(name string) generictables.MatchCriteria {
+	m.clauses = append(m.clauses, fmt.Sprintf("IPV saddr != @%s", LegalizeSetName(name)))
+	return m
+}
+
+func (m nftMatch) SourceIPPortSet(name string) generictables.MatchCriteria {
+	// IPPort sets include the IP, protocol, and port, in that order.
+	// Note that "th dport" is only compatible with protocols that have their destination port in
```

Comment:
```suggestion
	// Note that "th sport" is only compatible with protocols that have their destination port in
```

and in the next function. (The code is right, it's just the comment that's wrong.)